nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/data/explorer/StandardKustoIngestService.java

```diff
@@ -0,0 +1,404 @@
+        IngestionResult ingestionResult;
+        IngestionProperties ingestionProperties = new IngestionProperties(kustoIngestionRequest.getDatabaseName(),
+                kustoIngestionRequest.getTableName());
+
+        IngestionMapping.IngestionMappingKind ingestionMappingKind = setDataFormatAndMapping(kustoIngestionRequest.getDataFormat(), ingestionProperties);
+        if (StringUtils.isNotEmpty(kustoIngestionRequest.getMappingName()) && ingestionMappingKind != null) {
+            ingestionProperties.setIngestionMapping(kustoIngestionRequest.getMappingName(), ingestionMappingKind);
+        }
+
+        ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FAILURES_AND_SUCCESSES);
+        ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.TABLE);
+        ingestionProperties.setFlushImmediately(false);
+        ingestionProperties.setIgnoreFirstRecord(StringUtils.equalsIgnoreCase(kustoIngestionRequest.getIgnoreFirstRecord(), "YES"));
+
+        try {
+            if (kustoIngestionRequest.isStreamingEnabled()) {
+                ingestionResult = managedStreamingIngestClient.ingestFromStream(info, ingestionProperties);
+            } else {
+                ingestionResult = queuedIngestClient.ingestFromStream(info, ingestionProperties);
+            }
+            if (kustoIngestionRequest.pollOnIngestionStatus()) {
+                List<IngestionStatus> statuses = initializeKustoIngestionStatusAsPending();
+                long startTime = System.currentTimeMillis();
+                long timeoutMillis = kustoIngestionRequest.getIngestionStatusPollingTimeout().toMillis();
+                // Calculate the end time based on the timeout duration
+                long endTime = startTime + timeoutMillis;
+                while (System.currentTimeMillis() < endTime) {
+                    // Get the status of the ingestion operation
+                    List<IngestionStatus> statuses1 = ingestionResult.getIngestionStatusCollection();
+                    if (statuses1.get(0).status == OperationStatus.Succeeded
+                            || statuses1.get(0).status == OperationStatus.Failed
+                            || statuses1.get(0).status == OperationStatus.PartiallySucceeded) {
+                        statuses = statuses1;
+                        break;
+                    }
+                    // Sleep for before checking again
+                    Thread.sleep(kustoIngestionRequest.getIngestionStatusPollingInterval().toMillis());
+                }
+                // Check if the timeout has been exceeded
+                if (System.currentTimeMillis() - startTime >= timeoutMillis) {
+                    throw new ProcessException(String.format("Timeout of %s exceeded while waiting for ingestion status", kustoIngestionRequest.getIngestionStatusPollingTimeout()));
+                }
+                return KustoIngestionResult.fromString(statuses.get(0).status.toString());
```

Comment:
Recommend moving this block into a separate method named something like `pollIngestionStatus()`.